packages/@aws-cdk/aws-cloud9-alpha/lib/environment.ts

```diff
@@ -257,6 +257,26 @@ export class Owner {
     return { ownerArn: user.userArn };
   }
 
+  /**
+   * Make an IAM assumed role the environment owner
+   *
+   * @param accountId The AccountId of the target account
+   * @param roleName The name of the assumed role
+   */
+  public static assumedRole(accountId: string, roleName: string): Owner {
+    return { ownerArn: `arn:${cdk.Aws.PARTITION}:sts::${accountId}:assumed-role/${roleName}` };
+  }
+
+  /**
+   * Make an IAM federated user the environment owner
+   *
+   * @param accountId The AccountId of the target account
+   * @param userName The name of the federated user
+   */
+  public static federatedUser(accountId: string, userName: string): Owner {
+    return { ownerArn: `arn:${cdk.Aws.PARTITION}:sts::${accountId}:federated-user/${userName}` };
```

Comment:
Could you please tell me where did you find this `arn pattern`?

```
arn:${cdk.Aws.PARTITION}:sts::${accountId}:federated-user/${userName}
```

Reply:
The pattern is defined [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html) and matches the constraint documented in the [OwnerArn property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloud9-environmentec2.html#cfn-cloud9-environmentec2-ownerarn) in Cloud9s Cfn Resource Definiton